api/V1/types.proto

```diff
@@ -0,0 +1,43 @@
+syntax = "proto3";
+
+package celestia.node.v1.types;
+
+option go_package = "github.com/celestiaorg/celestia-node/api/v1/types";
+
+import "common.proto";
+import "proof.proto";
+
+// Core data types
+message Share {
+  bytes data = 1;
+}
+
+message SquareRange {
+  uint32 start = 1;
+  uint32 end = 2;
+}
+
+message Blob {
+  bytes data = 1;
+  optional bytes signer = 2;
+  bytes commitment = 3;
+}
```

Comment:
I've got your point the Blob is part of the namespace, and by submitting/getting the blob, the user already has it but I'd anyway make it part of the message for consistency.